wordpress-mobile/WordPress-Android
# Improved mocking for publishing posts

Since https://github.com/wordpress-mobile/WordPress-Android/pull/9835 has been merged, @JavonDavis has encountered problems with https://github.com/wordpress-mobile/WordPress-Android/pull/9835 since the requests the new tests used were not mocked.

I am working on making it easier to add new mocks, but in the meantime, this makes the necessary changes for the new test to pass with network mocking. I made these changes by proxying the app through Charles proxy and then updating the JSON mocks with the missing parts.

It makes the following changes:

- Add mocking for `/media/new` so that image uploading can be simulated. This will return a hardcoded image, but should be fine for testing. The image can be subsequently fetched with `/media/id`
- `taxonomies/post_tag/terms/`, `taxonomies/category/terms/` and `/categories` now return some sample categories/tags that can be used in testing.
- `posts/new/` is more flexible and now supports post title/content.

To test:

the best way to test this is to merge it into the `add/additional-publish-flow-test` branch:

1. Checkout the tests branch: `git checkout add/additional-publish-flow-test`.
2. Make a new branch so it doesn't mess up your local setup: `git checkout -b add/additional-publish-flow-test-mocked`.
3. Merge in the changes from this branch: `git merge publish-mocks` (resolve conflicts as needed).
4. run the new `testPublishFullPost` test and see that it now works with mocking enabled.

Update release notes:
